https://git.alpinelinux.org/aports/tree/community/qt5-qtwebengine/remove-glibc-check.patch

Qt checks if glibc is available and if not disables large part of Qt5WebEngine and thus cripples functionality.
However these parts work fine with Musl so there is no need to disable them. - Alpine

Just remove the check so it builds again. Since 5.15 is the last version of Qt to be released with this build system it'll be obsolete with Qt6 so there is no real point in fixing this upstream except adding to Qt5PatchCollection.

https://git.alpinelinux.org/aports/tree/community/qt5-qtwebengine/remove-glibc-check.patch
diff --git a/src/buildtools/config/support.pri b/src/buildtools/config/support.pri
index e7f869a1..de18523d 100644
--- a/src/buildtools/config/support.pri
+++ b/src/buildtools/config/support.pri
@@ -189,15 +189,6 @@ defineTest(qtwebengine_checkForHostPkgCfg) {
     return(true)
 }
 
-defineTest(qtwebengine_checkForGlibc) {
-    module = $$1
-    !qtConfig(webengine-system-glibc) {
-        qtwebengine_skipBuild("A suitable version >= 2.27 of libc required to build $${module} could not be found.")
-        return(false)
-    }
-    return(true)
-}
-
 defineTest(qtwebengine_checkForKhronos) {
     module = $$1
     !qtConfig(webengine-system-khr) {
diff --git a/src/buildtools/configure.json b/src/buildtools/configure.json
index 88d1790c..8623f6d7 100644
--- a/src/buildtools/configure.json
+++ b/src/buildtools/configure.json
@@ -264,18 +264,6 @@
             "label": "system gn",
             "type": "detectGn"
         },
-        "webengine-glibc": {
-            "label": "glibc > 2.16",
-            "type": "compile",
-            "test": {
-                "include": "features.h",
-                "tail": [
-                    "#if __GLIBC__ < 2 || __GLIBC_MINOR__ < 17",
-                    "#error glibc versions below 2.17 are not supported",
-                    "#endif"
-                ]
-            }
-        },
         "webengine-gperf": {
             "label": "gperf",
             "type": "detectGperf"
@@ -379,7 +367,6 @@
                          && (!config.sanitizer || features.webengine-sanitizer)
                          && (!config.linux || features.pkg-config)
                          && (!config.linux || features.webengine-host-pkg-config)
-                         && (!config.linux || features.webengine-system-glibc)
                          && (!config.linux || features.webengine-system-khr)
                          && (!config.linux || features.webengine-system-nss)
                          && (!config.linux || features.webengine-system-dbus)
@@ -517,11 +504,6 @@
             "condition": "config.unix && !config.darwin && libs.webengine-nss",
             "output": [ "privateFeature" ]
         },
-        "webengine-system-glibc": {
-            "label": "glibc",
-            "condition": "config.linux && tests.webengine-glibc",
-            "output": [ "privateFeature" ]
-        },
         "webengine-system-x11" : {
             "label": "x11",
             "condition": "config.unix && libs.webengine-x11",
@@ -782,8 +764,7 @@
                         "webengine-system-fontconfig",
                         "webengine-system-dbus",
                         "webengine-system-nss",
-                        "webengine-system-khr",
-                        "webengine-system-glibc"
+                        "webengine-system-khr"
                     ]
                 },
                 {
